src/CommonLib/hostfxr_utility.cpp

```diff
@@ -235,25 +205,86 @@ HOSTFXR_UTILITY::GetPortableHostfxrLocation(
     {
         goto Finished;
     }
-    hr = struHostfxrPath->Append(L"\\");
-    if (FAILED(hr))
+
+    if (FAILED(hr = struHostFxrPath.Append(L"\\"))
+        || FAILED(hr = struHostFxrPath.Append(strHighestDotnetVersion.QueryStr()))
+        || FAILED(hr = struHostFxrPath.Append(L"\\hostfxr.dll")))
     {
         goto Finished;
     }
 
-    hr = struHostfxrPath->Append(strHighestDotnetVersion.QueryStr());
-    if (FAILED(hr))
+    hFileHandle = UTILITY::CheckIfFileExists(&struHostFxrPath);
+
+    if (hFileHandle == INVALID_HANDLE_VALUE)
+    {
+        hr = ERROR_FILE_INVALID;
+        goto Finished;
+    }
+    
+    if (FAILED(hr = GetArguments(pConfig->QueryArguments(), &strDotnetExeLocation, pConfig)))
     {
         goto Finished;
+    }
 
+    if (FAILED(pConfig->QueryHostFxrLocation()->Copy(struHostFxrPath)))
+    {
+        goto Finished;
     }
 
-    hr = struHostfxrPath->Append(L"\\hostfxr.dll");
-    if (FAILED(hr))
+Finished:
+
+    if (hFileHandle != INVALID_HANDLE_VALUE)
     {
+        CloseHandle(hFileHandle);
+    }
+    return hr;
+}
+
+//
+// Forms the argument list in HOSTFXR_PARAMETERS.
+// Sets the ArgCount and Arguments.
+// Arg structure:
+// argv[0] = Path to exe activating hostfxr.
+// argv[1] = L"exec"
+// argv[2] = first argument specified in the arguments portion of aspnetcore config. 
+// 
+HRESULT
+HOSTFXR_UTILITY::GetArguments(
+    STRU* struArgumentsFromConfig, 
+    STRU* pstruExePath, 
+    ASPNETCORE_CONFIG* pConfig
+)
+{
+    HRESULT     hr = S_OK;
+    INT         argc = 0;
+    PCWSTR*     argv = NULL;
+    LPWSTR*     pwzArgs = NULL;
+
+    pwzArgs = CommandLineToArgvW(struArgumentsFromConfig->QueryStr(), &argc);
+
+    argv = new PCWSTR[argc + 2];
+    if (argv == NULL)
+    {
+        hr = E_OUTOFMEMORY;
         goto Finished;
     }
 
+    argv[0] = SysAllocString(pstruExePath->QueryStr());
+    argv[1] = SysAllocString(L"exec");
+
+    for (INT i = 0; i < argc; i++)
+    {
+        argv[i + 2] = SysAllocString(pwzArgs[i]);
+    }
+
+    *pConfig->QueryHostFxrArgCount() = argc + 2;
```

Comment:
It's not a good practice to do change the returned value of Query function. You should have a Set function. Query function should return const value